libc-top-half/musl/src/thread/wasm64/wasi_thread_start.s

```diff
@@ -0,0 +1,48 @@
+	.text
+
+	.export_name	wasi_thread_start, wasi_thread_start
+
+	.globaltype	__stack_pointer, i64
```

Comment:
Rather than duplicating this file perhaps just move it to `src/thread/wasm/` and then rename it to `.S` so you can do something like this:

```
#ifdef __wasm64__
#define PTR i64
#else
#define PTR i32
#endif
```

Reply:
The __wasm64__ macro does not exist in the asm compiler.

Reply:
(note you need to use the uppercase `.S` filename to get the pre-processor to kick in)

Reply:
i am trying to merge them

Reply:
how does macro work in .s?

`
#ifdef BUILDWASM64
#define PTR i64
#define PTRSIZE 8
#else
#define PTR i32
#define PTRSIZE 4
#endif

	.globaltype	__stack_pointer, PTR
`

`libc-top-half/musl/src/thread/wasm32/wasi_thread_start.s:13:31: error: Unknown type in .globaltype directive: PTR
        .globaltype     __stack_pointer, PTR
                                         ^
`

Reply:
clang --version
clang version 18.0.0 (https://github.com/llvm/llvm-project.git 980e024bf795145949acb0e2aacf808c41a5c531)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/local/bin

`
#ifdef __wasm64__
WTF??
#endif
`

`
clang --target=wasm32-wasi-threads a.s
a.s:2:1: error: invalid instruction
WTF??
^
`

Reply:
fixed